flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContext.java

```diff
@@ -289,6 +290,20 @@ public void signalNoMoreSplits(int subtask) {
                 "Failed to send 'NoMoreSplits' to reader " + subtask);
     }
 
+    @Override
+    public void signalIntermediateNoMoreSplits(int subtask) {
+        checkSubtaskIndex(subtask);
+
+        // It's an intermediate noMoreSplit event, notify subtask to deal with this event.
+        callInCoordinatorThread(
+                () -> {
+                    subtaskHasNoMoreSplits[subtask] = false;
```

Comment:
Why is it needed to be set to false?

Reply:
In my understanding, it should be `false` initially and set to `true` when on `signalNoMoreSplits(int)`. When failover happens, the value will be reset to `false`. Seems there is no need for `signalIntermediateNoMoreSplits(int)` to do the reset.

Reply:
yes, no need to set false. i delete it. thanks.